### PR TITLE
Milestone 3: Set up associations

### PR DESCRIPTION
**Pull request by BenMKT**
## Project scope:

Finish creating the remaining classes for our school library and create the associations between them thus;

- Create a class `Classroom`.
- Create a class `Book`.
- Create a class `Rental`.

### CODE REVIEW

File: class_person.rb [hunk 1]
```diff
@@ -2,12 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
+  attr_reader :id, :rentals
```
Comment (LionRouge1) [inline class_person.rb]: - [x] Great job implementing the `Person` class and setting up **Setter** and **Getter** for the instance variables `@name` and `@age` :clap: However, it's important that we set up both **Setter** and **Getter** for the `@rentals` variable as well. That will help to manage a person's rental outside the `Person` class, like adding a rental to a person in the `Rental` class. So Kindly set up both **Getter** and **Setter** for the `@rentals` variable as you did for the `@name` and `@age`

File: class_person.rb [hunk 2]
```diff
@@ -18,6 +21,12 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
+  def add_rental(book, date)
+    rental = Rental.new(date, book, self)
+    @rentals.push(rental)
+    rental
```
Comment (LionRouge1) [inline class_person.rb]: - [x] Please note that when you create a new rental, it automatically adds itself to the person's rentals thanks to this code `person.rentals << self` in the `Rental` class at line `9`. That means we don't need to push the new rental to @rentals after it has been created. Therefore I suggest you remove this code from the `add_rental` method. 
  Example:
```ruby
def add_rental(book, date)
  Rental.new(date, book, self)
end
```

File: class_book.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    rental = Rental.new(date, self, person)
+    @rentals.push(rental)
+    rental
```
Comment (LionRouge1) [inline class_book.rb]: - [x] Same for the book as pointed out :point_up:  in the previous comment. When a new Rental is created it adds itself to the book's rentals, thanks to this code `book.rentals << self` in the `Rental` class at line `8`. So pushing the new rental to the book's rentals is adding the same rental twice. Therefore I suggest you remove this code from the `add_rental` method.
  Example:
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @BenMKT, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Create the remaining classes with their instance variable :heavy_check_mark: 
- Create `many-to-many` and `has-many/belongs-to` relationships between two classes :heavy_check_mark: 
- All linters passed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

[OPTIONAL]
- [x] It is recommended in ruby best practices to name a file after the class name with all the characters in minuscule (`class Person` => `person.rb`, `class Rental` => `rental.rb`, ...). Please note that you don't need to prefix each file with **`class`**, that makes your project look unprofessional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class_person.rb (hunk 2 above)
Comment (BenMKT) [reply class_person.rb]: - Remove unnecessary code from `add-rental` method to resolve bug.

File: class_person.rb (hunk 1 above)
Comment (BenMKT) [reply class_person.rb]: - Set up both Getter and Setter for the `@rentals` variable

File: class_book.rb (hunk 3 above)
Comment (BenMKT) [reply class_book.rb]: - Remove unnecessary code from `add-rental` method to resolve bug.

**Review by Deepakdanger — APPROVED** [review]
Hi @BenMKT , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**